### ISSUE TITLE
[Federation] Schema built with buildFederatedSchema does not run custom scalar parseLiteral

### ISSUE THREAD

**Issue by hannesj** [apollographql/apollo-server#2832]
It seems custom scalars' parseLiteral is not run, when using them as inputs in a schema where federation is enabled.

I created a small repro in https://github.com/hannesj/federation-migration-example.

You can see the issue by running the server and sending the following query: 

`{ reservation(id: "UmVzZXJ2YXRpb24=:MQ==") { id } }`

When running `npm start` in the `before-migration/reservations` directory and running the above query, the following is output, and the resolver gets the correctly typed input:

```
{
  kind: 'StringValue',
  value: 'UmVzZXJ2YXRpb24=:MQ==',
  block: false,
  loc: { start: 20, end: 43 }
}
TypedID { type: 'Reservation', id: '1' }
```

When running `npm start` in the `after-migration/reservations` directory and running the above query, the following is output, and the resolver receives an unparsed string as input:

```
UmVzZXJ2YXRpb24=:MQ==
```

**Comment by hannesj** [comment]
It seems the issue is actually in `apollo-tooling`. [Here](https://github.com/apollographql/apollo-tooling/blob/25838b6a3220ff8d71fb74125aecfa52beeb7f9c/packages/apollo-graphql/src/schema/buildSchemaFromSDL.ts#L219), the `parseLiteral`, `parseValue` and `serialize` functions should be copied from the `GraphQLScalarType` in the resolvers to the type in the schema

**Comment by jbaxleyiii** [comment]
@hannesj thanks for the reproduction and pointing to the change! Do you have any interest in making that change in a PR? We would be happy to help if you have interest!

**Comment by hannesj** [comment]
Sure, I'll take a look.

**Comment by hannesj** [comment]
@jbaxleyiii Done in https://github.com/apollographql/apollo-tooling/pull/1343

**Comment by hannesj** [comment]
Released in `apollo-graphql@0.3.3`

**Comment by nevir** [comment]
Note that this is inconsistent vs `makeExecutableSchema` of old:

If you were providing "resolvers" for scalar types via `makeExecutableSchema`, it expected `__` prefixed special functions.

```ts
const myScalarResolvers = {
  __parseValue(…) { … },
  __parseLiteral(…) { … },
  __serialize(…) { … },
};
```

Whereas, with `buildFederatedSchema`, we're assigning directly. So you'll need to convert to:

```ts
const myScalarResolvers = {
  parseValue(…) { … },
  parseLiteral(…) { … },
  serialize(…) { … },
};
```
